SECURITY.md

```diff
@@ -1,34 +1,93 @@
 - When you suspect a potential vulnerability but you are not sure if or how it would impact PowSyBl.
 - When you know or suspect a vulnerability on another project that is used by PowSyBl.
 
-## Security on new code
+## Supported versions
+
+The available compatible releases are listed in [powsybl-dependencies](https://github.com/powsybl/powsybl-dependencies). In case of vulnerability fix, the main branches are first fixed. 
+Only the most recent version will include the patch (through the usual release train or through a patch released outside the release train, depending on the severity).
+
+## PowSyBl security requirements
+
+### Be harmless on user environment
+
+The PowSyBl code is executed on users environments. Using the PowSyBl code must not cause damage to the users environments.
+
+### Be safe with user data
+The PowSyBl code is executed on users data. Using the PowSyBl code must not compromise the data.
+
+## Threat model
+
+### Take advantage of known vulnerabilities / bugs / bad design principles
+
+An attacker interacting with user systems powered by PowSyBl may try to use known vulnerabilities from external dependencies or take advantage of bugs or bad design principles.
+
+### Making a malicious Pull Request
+
+An attacker may try to push a malicious pull request, deliberatly introducing security vulnerabilities.
+
+## Unapplicable threats
+
+### Unapplicable: attacks on infrastructure
+
+PowSyBl has no calculation or database servers. All calculations are run on a user environment and databases belong to users too.
 
-### Sonarcloud analysis
+## Implemented measures
+
+In order to meet the PowSyBl security requirements and avoid possible threats, the PowSyBl team has several tools and methods listed below.
+
+### Hunting down known security flaws
+
+#### Regular checks on dependency vulnerabilities
+
+Checks on known dependency vulnerabilities are automatically performed through GitHub actions. 
+
+#### Recent Java version
+
+We support recent Java versions so that users can benefit from the latest security improvements.
+
+#### Controlled dependency update
+
+We pay attention to keep our dependency versions up-to-date and also make sure the new versions are safe. This allows PowSyBl to benefit from the latest security patches, while ensuring a protection against supply-chain attacks.
+
+#### Sonarcloud analysis
 
 For new pull requests, we rely on Sonarcloud analysis that has [security-related rules](https://docs.sonarcloud.io/digging-deeper/security-related-rules/). The pull request cannot be merged if any vulnerability or hotspot rule is violated.
 
-###  PowSyBl governance
+#### Open source and community
+
+We encourage our users to report and investigate any vulnerability according to the recommendations of this document.
+
+#### Developers best practices
+
+We encourage contributors to use strong security practices (GPG keys, 2FA, safe and up-to-date operating systems, development tools and web browsers, etc.).
```

Comment:
2FA is indeed mandatory.

Reply:
Yes. We thought we would add it anyway because we encouraged it before it was mandatory. 
I will remove it however, since it may seem that we are not aware that it is a GitHub requirement.